torch/csrc/jit/serialization/export_module.cpp

```diff
@@ -522,21 +577,55 @@ class ScriptModuleSerializer {
       // If exception ocurrs we have a model with debug map that can be
       // used to symbolicate debug handles
       writeArchive("mobile_debug_handles", debug_info_telements);
+      static constexpr size_t kMinToCompress = 200;
+      // For delegated backends get source ranges that are in the debug info
+      // map. Since delegated backend replace original module with lowered
+      // module we will not serialize original module's code which is what would
+      // have contained source range. Since we dont have that anymore, extract
+      // source ranges out of delegated module and store in a separate archive.
+      // Note that we must do this first because in order to serialize inlined
+      // CS appropriate source_range_tags must have been generated.
+      auto backend_source_range_records = getBackendSourceRanges(module);
```

Comment:
Just thinking out load: maybe at some point it may make sense to have separate .debug_pkl per backend type.